### PR TITLE
Audio loader

### PR DESCRIPTION
**Pull request by joesus**
## Checklist

- [x] I've read the [Contributing Guidelines](CONTRIBUTING.md) and the [Code of Conduct](CODE_OF_CONDUCT.md)
- [x] I've completed the [Contributor License Agreement](https://developers.facebook.com/opensource/cla)
- [x] I've ensured that all existing tests pass and added tests (when/where necessary)
- [x] I've ensured that my code lints properly: (`swiftlint && swiftlint autocorrect --format`)
- [x] I've updated the documentation (when/where necessary) and [Changelog](CHANGELOG.md) (when/where necessary)
- [x] I've added the proper label to this pull request (e.g. `bug` for bug fixes)

## Pull Request Details

Adds AudioResourceLoader for saving Data to disk for use in creating system sounds. This allows us to avoid having to bundle sound assets with the sdk as well as helping avoid the overhead of AVAudioPlayer.

## Test Plan

Unit tests should pass.

### CODE REVIEW

File: Sources/Core/AudioResourceLoader.swift [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    }
+  }
+
+  mutating func play(resource: AudioResource.Type) throws {
+    let systemSoundIdentifier = try cache[resource.name] ?? (try load(resource: resource))
+
+    audioService.playSystemSound(with: systemSoundIdentifier)
+  }
+
+  enum SystemError: FBError {
```
Comment (codytwinton) [inline Sources/Core/AudioResourceLoader.swift]: I think we should probably come up with a less generic name for these errors.

```suggestion
  enum AudioSystemError: FBError {
```

File: Sources/Core/Extensions/FileManager+Extensions.swift [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    in domain: FileManager.SearchPathDomainMask,
+    appropriateFor url: URL?,
+    create shouldCreate: Bool
+    ) throws -> URL
+
+  func createDirectory(
+    at url: URL,
+    withIntermediateDirectories createIntermediates: Bool,
+    // swiftlint:disable:next discouraged_optional_collection
+    attributes: [FileAttributeKey: Any]?
```
Comment (codytwinton) [inline Sources/Core/Extensions/FileManager+Extensions.swift]: Why optional?

```suggestion
    attributes: [FileAttributeKey: Any] = [:]
```

Comment (joesus) [reply Sources/Core/Extensions/FileManager+Extensions.swift]: Unfortunately this mirrors the FileManager api so you'll have to ask Apple. 🙄

File: Sources/Core/AudioResourceLoader.swift (hunk 1 above)
Comment (joesus) [reply Sources/Core/AudioResourceLoader.swift]: Agreed.

**Pull request merged**